qiskit/quantum_info/states/statevector.py

```diff
@@ -242,10 +247,16 @@ def evolve(self, other, qargs=None):
         return Statevector._evolve_operator(ret, other, qargs=qargs)
 
     def equiv(self, other, rtol=None, atol=None):
-        """Return True if statevectors are equivalent up to global phase.
+        """Return True if the statevector of other and self are equivalent up to global phase.
+
+        .. note::
+
+            If ``other`` is a ``QuantumCircuit``, the statevector is constructed by assuming that
+            all qubits in the circuit are initialized to the zero state.
 
         Args:
-            other (Statevector): a statevector object.
+            other (Statevector or vector_like or QuantumCircuit): an object from which a
```

Comment:
I would just leave the extra types out
```suggestion
            other (Statevector): a statevector or object that can be used to initialize a statevector.
```

Reply:
done in 5c59880